src/savegame.cpp

```diff
@@ -451,6 +433,11 @@ bool savegame::save_game(const std::string& filename)
 		if (filename_.empty())
 			filename_ = filename;
 
+		time_t t = std::time(nullptr);
+		tm tm = *std::localtime(&t);
+		auto time = std::put_time(&tm, "%Y%m%d-%H%M%S");
+		filename_ = (formatter() << filename_ << '_' << time).str();
```

Comment:
This appends the date each time this function is called regardless if the saved filename in the member variable is used or not. That does not seem like a good idea? Seems like the previous timestamp needs to be replaced.

Reply:
I tested re-saving the game/replay multiple times, and it's not an issue, since the file name gets re-created from scratch each time anyway.

Reply:
What is the if empty check above for then? Dead code?

Reply:
I believe so.  The two places where `save_game()` is called both set `filename_` to a value.  That check also doesn't account for both `filename_` and `filename` being empty, even though the `filename` argument defaults to being empty.

Reply:
Ok, just confusing code that should be cleaned up at some point then. (and maybe a quick history check what the intention was...)